src/shared/sam/localLambdaRunner.ts

```diff
@@ -113,8 +113,8 @@ export async function makeInputTemplate(
     if (config.lambda?.timeoutSec) {
         newTemplate = newTemplate.withTimeout(config.lambda?.timeoutSec)
     }
-    if (config.invokeTarget.architecture) {
-        newTemplate = newTemplate.withArchitectures([config.invokeTarget.architecture])
+    if (config.architecture) {
```

Comment:
This should still read from `invokeTarget`. Generated templates only make sense in the context of 'code' targets